buildstockbatch/base.py

```diff
@@ -702,6 +705,61 @@ def validate_resstock_or_comstock_version(project_file):
 
         return True
 
+    @staticmethod
+    def validate_number_of_options(project_file):
+        """Checks that there aren't more options than are allowed in the ApplyUpgrade measure.
+
+        :param project_file: path to project file
+        :type project_file: str
+        :raises ValidationError: if there are more options defined than there are in the measure.
+        :return: whether validation passes or not
+        :rtype: bool
+        """
+        cfg = get_project_configuration(project_file)
+        measure_xml_filename = os.path.join(
+            cfg["buildstock_directory"], "measures", "ApplyUpgrade", "measure.xml"
+        )
+        if os.path.exists(measure_xml_filename):
+            measure_xml_tree = objectify.parse(measure_xml_filename)
+            measure_xml = measure_xml_tree.getroot()
+            n_options_in_measure = 0
+            n_costs_per_option_in_measure = 0
+            for argument in measure_xml.arguments.argument:
+                m_option = re.match(r"^option_(\d+)$", str(argument.name))
+                if m_option:
+                    option_number = int(m_option.group(1))
+                    n_options_in_measure = max(option_number, n_options_in_measure)
+                m_costs = re.match(r"^option_(\d+)_cost_(\d+)_value", str(argument.name))
+                if m_costs:
+                    cost_number = int(m_costs.group(2))
+                    n_costs_per_option_in_measure = max(cost_number, n_costs_per_option_in_measure)
```

Comment:
Here I'm counting up the number of options and costs per option in the ApplyUpgrade measure.

Reply:
Would it speed things up to not check every option's number of costs (since we know it's the same number for every option)?

Reply:
This already runs pretty fast enough. It seems like an unnecessary optimization to me. Since we have to loop over all the arguments anyway, it's just as easy and simpler to check on each one.